geomstats/geometry/discrete_curves.py

```diff
@@ -842,3 +842,340 @@ def g_criterion(srv, srv_norms):
+        See [KN2018]_ for details.
+
+        Parameters
+        ----------
+        f : array-like, shape=[..., n_sampling_points - 1, ambient_dim]
+            F_transform of the curve.
+
+        Returns
+        -------
+        curve : array-like, shape=[..., n_sampling_points, ambient_dim]
+            Discrete curve.
+        """
+        n_sampling_points = f.shape[-2]
+        norms, args = self.cartesian_to_polar(f)
+        curve_x = [0]
+        curve_y = [0]
+
+        for i in range(n_sampling_points):
+            x = curve_x[-1]
+            x += norms[i]**2 * gs.cos(2 * self.b / self.a * args[i])
+            curve_x.append(x)
+            y = curve_y[-1]
+            y += norms[i]**2 * gs.sin(2 * self.b / self.a * args[i])
+            curve_y.append(y)
+
+        curve_x = gs.array(curve_x)
+        curve_y = gs.array(curve_y)
+        curve = gs.vstack((curve_x, curve_y)).T
+        curve = 1 / (4 * self.b**2) * curve + starting_point
+
+        return curve
+
+    def dist(self, curve_1, curve_2, rescaled=False):
+        """Compute the geodesic distance between two curves.
+
+        The two F_transforms are computed with corrected arguments
+        before taking the L2 distance between them.
+        See [KN2018]_ for details.
+
+        Parameters
+        ----------
+        curve_1 : array-like, shape=[..., n_sampling_points, ambient_dim]
+            Discrete curve.
+        curve_2 : array-like, shape=[..., n_sampling_points, ambient_dim]
+            Discrete curve.
+
+        Returns
+        -------
+        dist : [...]
+            Geodesic distance between the curves.
+        """
+        velocity_1 = curve_1[1:] - curve_1[:-1]
+        velocity_2 = curve_2[1:] - curve_2[:-1]
+
+        speed_1, arg_1 = self.cartesian_to_polar(velocity_1)
+        speed_2, arg_2 = self.cartesian_to_polar(velocity_2)
+
+        n_sampling_points = arg_1.shape[-1]
+
+        for i in range(n_sampling_points):
+            if arg_2[i] - arg_1[i] > 2 * gs.pi * 0.9:
+                for j in range(i, n_sampling_points):
+                    arg_2[j] -= 2 * gs.pi
+            elif arg_2[i] - arg_1[i] < -2 * gs.pi * 0.9:
+                for j in range(i, n_sampling_points):
+                    arg_2[j] += 2 * gs.pi
+
+        f_1_args = arg_1 * self.a / (2 * self.b)
+        f_2_args = arg_2 * self.a / (2 * self.b)
+
+        f_1_norms = 2 * self.b * gs.sqrt(speed_1)
+        f_2_norms = 2 * self.b * gs.sqrt(speed_2)
+
+        f_1 = self.polar_to_cartesian(f_1_norms, f_1_args)
+        f_2 = self.polar_to_cartesian(f_2_norms, f_2_args)
+
+        l2_prod = self.l2_metric(n_sampling_points).inner_product
+        l2_dist = self.l2_metric(n_sampling_points).dist
+
+        if rescaled:
+            cosine = l2_prod(f_1, f_2) / (4 * self.b**2)
+            distance = 2 * self.b * gs.arccos(gs.clip(cosine, -1, 1))
+        else:
+            distance = l2_dist(f_1, f_2)
+
+        return distance
+
+
+class ElasticCurves(Manifold):
```

Comment:
Why do we need a new class here? How about: adding the elastic metric as an attribute of the existing classes of `DiscreteCurves` and/or `ClosedDiscreteCurves` (I don't remember if it would work in the closed curves case?).

Reply:
Indeed, I just thought DiscreteCurves was reserved for the srv metric.

Reply:
Actually I'm not sure of how I should add the elastic metric as an attribute of `DiscreteCurves` since it only works for planar discrete curves (i.e. `ambient_manifold = Euclidean(dim=2)`.

Reply:
You could add `if dim == 2: self.elastic_metric = ...`?